### DISCUSSION TITLE
What to do if the feed has problems?

### DISCUSSION THREAD

**Discussion by math-GH** [FreshRSS/FreshRSS#6287] [Q&A]
Let's collect here some step what to do if the feed does not work as expected:

Here my first thoughts:
1) Link to the feed
2) describe the error. Do you have any error message?
3) FreshRSS log entries?
4) check validation via https://validator.w3.org/feed/ (it is the same link as it is given in the feed configs (button `Check the validity of the feed`
5) if possible try curl

**Comment by Frenzie** [comment]
curl plus curl with a FreshRSS user agent

**Reply by math-GH** [reply]
FreshRSS user agent or browser user agent?

**Reply by Frenzie** [reply]
Definitely the former, but the latter doesn't hurt. A curl user agent and a browser user agent don't show you what FreshRSS gets back after all.

In fact we can probably automatically generate a curl command in settings → about or installation check or something, or (hidden by default) on the feed itself. That is, simply `curl -A "$user_agent" $url`.

**Comment by math-GH** [comment]
Which concrete curl commands should be tested?

**Comment by Alkarex** [comment]
With cURL, a number of variants may be tried, starting with various user agents (most common problem) (we should provide some suggestions).

First from the same server than where FreshRSS is located, if possible, and then from other IPs (if the responses are different, then there is an IP-based filter for which the solution might be a proxy).

Then try cURL with various IP versions (IPv4 / IPv6) via `-4` and `-6`, and various versions of HTTP with `--http1.0`, `--http1.1`, `--http2`, `--http3`.

Then with [curl-impersonate](https://github.com/lwthiker/curl-impersonate) (trying to impersonate Chrome and/or Firefox) (e.g. easy with Docker):

```
docker run --rm lwthiker/curl-impersonate:0.6-chrome curl_chrome101 'https://www.tenforums.com/external.php?type=RSS2'
```

**Comment by math-GH** [comment]
If cloudflare is breaking the access to the feeds and pages:
see #4323

**Reply by Alkarex** [reply]
We should highlight the possibility of making a feed proxy in Feedburner for blocked feeds: https://feedburner.google.com/

**Reply by Frenzie** [reply]
Some pointers (copied from <https://github.com/FreshRSS/FreshRSS/issues/7454#issuecomment-2748544730>)

* check with a tool like cURL or HTTPie, not with a browser (non-browser user agent, no JavaScript)
* if that doesn't work, there's your answer: a cause like CF should be fairly self-evident at this point, especially with `-v`
* if it works, check cURL with a FreshRSS user agent, and conversely, you can try FreshRSS with a browser user agent
* if that also works, don't forget to check from the server, not just from your local device on another network
* if there's still a discrepancy, check the version of the cURL library used by PHP (sometimes it can be astonishingly outdated)
